test/jasmine/tests/legend_test.js

```diff
@@ -882,4 +884,68 @@ describe('legend interaction', function() {
             .then(done);
         });
     });
+
+    describe('editable mode interactions', function() {
+        var gd;
+        var mock = {
+            data: [{
+                x: [1, 2, 3],
+                y: [5, 4, 3]
+            }, {
+                x: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14],
+                y: [1, 3, 2, 4, 3, 5, 4, 6, 5, 7, 6, 8, 7, 9],
+                transforms: [{
+                    type: 'groupby',
+                    groups: [1, 2, 1, 2, 3, 4, 3, 4, 5, 6, 5, 6, 7, 8]
+                }]
+            }],
+            config: {editable: true}
+        };
+
+        beforeEach(function(done) {
+            gd = createGraphDiv();
+            Plotly.plot(gd, Lib.extendDeep({}, mock)).then(done);
+        });
+
+        afterEach(destroyGraphDiv);
+
+        function _setValue(index, str) {
+            var item = d3.selectAll('text.legendtext')[0][index || 0];
+            item.dispatchEvent(new MouseEvent('click'));
+            return delay(20)().then(function() {
+                var input = d3.select('.plugin-editable.editable');
+                input.text(str);
+                input.node().dispatchEvent(new KeyboardEvent('blur'));
+            }).then(delay(20));
+        }
+
+        it('sets and unsets trace group names', function(done) {
```

Comment:
It's testing a fair amount of feature actions which isn't the *best*, but it's not bad for end-to-end.